flake8_bandit.py

```diff
@@ -106,14 +106,26 @@ def _check_source(self):
         ):
             return []
 
-        bnv = BanditNodeVisitor(
-            self.filename,
-            BanditMetaAst(),
-            BanditTestSet(BanditConfig(), profile=config.profile),
-            False,
-            [],
-            Metrics(),
-        )
+        try:
+            bnv = BanditNodeVisitor(
+                fname=self.filename,
+                fdata=None,
+                metaast=BanditMetaAst(),
+                testset=BanditTestSet(BanditConfig(), profile=config.profile),
+                debug=False,
+                nosec_lines=[],
+                metrics=Metrics(),
+            )
+        except TypeError:
+            # bandit < 1.7.3 (https://github.com/tylerwince/flake8-bandit/issues/21)
```

Comment:
Personally I would use the [other PR approach](https://github.com/tylerwince/flake8-bandit/pull/22/files), no real need to create this compatibility code. It just makes the code less maintainable and not a huge gain...

Reply:
I'm not sure if we should pin versions in a package such as this one ...

Reply:
I agree @staticdev , the `TypeError` catch here makes an assumption that `BanditNodeVisitor` won't change again in future. We may just end up with a new generic `TypeError` being caused by the *older* call below, when in fact the root cause would be something else entirely.

Better to make two updates in this package, I feel:

- A patch update for `flake8-bandit` (to v2.1.3) that sets the `bandit` version in `setup.py` to `<1.7.3`
- A minor update (to v2.2.0) that implements the backwards-incompatible change in #22 , and setting `bandit>=1.7.3`